test/expr/cast/nil_value_to_optional.swift

```diff
@@ -30,3 +35,13 @@ _ = dopt == nil
 _ = diuopt == nil
 _ = diuopt is ExpressibleByNilLiteral // expected-warning {{'is' test is always true}}
 _ = produceD() is ExpressibleByNilLiteral // expected-warning {{'is' test is always true}}
+
+enum E {
+  case none
+}
+func test(_ e: E) {
+  _ = e == .none
+  _ = e == E.none
+  _ = e == Optional.none // expected-warning {{comparing non-optional value of type 'E' to 'nil' or 'Optional.none' always returns false}}
+  _ = e == E?.none // expected-warning {{comparing non-optional value of type 'E' to 'nil' or 'Optional.none' always returns false}}
```

Comment:
These need to be updated too.

Reply:
good catch, thanks! I'm running the tests locally, could you trigger a smoke test run on the PR?

Reply:
There was one related test failure so I pushed a fix. Related tests pass for me now when I run them locally, mind triggering another smoke test? (thanks 😄)